Mathlib/GroupTheory/Perm/Sign.lean

```diff
@@ -112,14 +113,34 @@ theorem swap_induction_on [Finite α] {P : Perm α → Prop} (f : Perm α) :
         (ih _ ⟨rfl, fun v hv => hl.2 _ (List.mem_cons_of_mem _ hv)⟩ h1 hmul_swap)
 #align equiv.perm.swap_induction_on Equiv.Perm.swap_induction_on
 
-theorem closure_isSwap [Finite α] : Subgroup.closure { σ : Perm α | IsSwap σ } = ⊤ := by
-  cases nonempty_fintype α
-  refine' eq_top_iff.mpr fun x _ => _
-  obtain ⟨h1, h2⟩ := Subtype.mem (truncSwapFactors x).out
-  rw [← h1]
-  exact Subgroup.list_prod_mem _ fun y hy => Subgroup.subset_closure (h2 y hy)
+theorem mclosure_isSwap [Finite α] : Submonoid.closure { σ : Perm α | IsSwap σ } = ⊤ :=
+  eq_top_iff.mpr fun x _ ↦ by
+    cases nonempty_fintype α
+    obtain ⟨h1, h2⟩ := Subtype.mem (truncSwapFactors x).out
+    rw [← h1]
+    exact Submonoid.list_prod_mem _ fun y hy ↦ Submonoid.subset_closure (h2 y hy)
```

Comment:
Is there a reason to reorder this proof? I'm not convinced the new code is necessarily better.

Reply:
I think I originally wanted to golf it but didn't end up finding any golf ... let me change it back.